x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings_by_resource/resource_findings/use_resource_findings.ts

```diff
@@ -34,10 +34,12 @@ type ResourceFindingsResponse = IKibanaSearchResponse<
   estypes.SearchResponse<CspFinding, ResourceFindingsResponseAggs>
 >;
 
-export type ResourceFindingsResponseAggs = Record<
-  'count' | 'clusterId' | 'resourceSubType' | 'resourceName',
-  estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>
->;
+export interface ResourceFindingsResponseAggs {
+  count?: estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>;
+  clusterId?: estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>;
+  resourceSubType?: estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>;
+  resourceName?: estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>;
+}
```

Comment:
if we add  `| undefined` to the previous types we could keep the `Record` which seems a better option here given all keys share the same value type. 

```suggestion
export type ResourceFindingsResponseAggs = Record<
  'count' | 'clusterId' | 'resourceSubType' | 'resourceName',
  estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys | undefined>
>;
```

Reply:
@orouz  I didn't see comments until I approved. I'll create an additional PR to address changes clean up.